deps/secure-socket/biowrap.lua

```diff
@@ -48,6 +48,10 @@ return function (ctx, isServer, socket, handshakeComplete, servername)
 
   local function handshake(callback)
     if ssl:handshake() then
+      local cert = ssl:peer()
+      if not cert:check_host(servername) then
+        return error("The server hostname does not match the certificate's domain")
+      end
```

Comment:
This whole check should be moved to after peer verification has succeeded, it's unnecessary work if we don't even trust the certificate.

Reply:
Can't we also argue that doing the actual certificate validation is more work if the certificate isn't even who it claims to be? If identity check takes more CPU time than peer validation then I agree.

Reply:
getpeerverification is essentially a no-op, it just returns the stored verification result. Actual peer verification happens during the ssl handshake.